### ISSUE TITLE
Expose if loading failed because of an error or because of a blocker

### ISSUE THREAD

**Issue by danrr** [muicss/loadjs#50]
Hi! I'm using loadjs (great library btw) and I'm seeing some infrequent loading failures in the logs. I'm trying to establish if it's caused by something blocking the scripts or if they are failing to load for some other reason. 

The `loadFiles` function already has logic to detect blocked scripts but it does not expose this information, only which scripts were blocked, if any. I'd like to be able to add this to the logs to better diagnose the what causing the failed script loads. 

What do you think?

**Comment by amorey** [comment]
Are you trying to determine if a script was blocked by an ad blocker or something else?

**Comment by danrr** [comment]
Yes, I'd like to know if it was blocked by an ad blocker. Particularly, the logic here https://github.com/muicss/loadjs/blob/master/src/loadjs.js#L166.

**Comment by amorey** [comment]
It should be easy to modify the library to return that information but I'm not sure we can reliably detect blocked scripts so I've actually been thinking about removing that code from the library altogether (unless we can fix it). Here's the github issue:
https://github.com/muicss/loadjs/issues/47

**Comment by danrr** [comment]
OK, I see. I have looked at issue before posting but didn't know you were looking to remove the detection. Has there been any additional testing to verify that the detection isn't working or if it's specific browser/ad blocker combos that it doesn't work for?

**Comment by amorey** [comment]
I haven't had a chance to look too closely at the issue yet. Do you know of any libraries that can detect blocked scripts that work across browsers/adblocker?

**Comment by danrr** [comment]
Sorry, I don't know of any off the top of my head.

**Comment by danrr** [comment]
I might have a play with different adblockers and browsers when I get some time, and report back with any findings.

**Comment by amorey** [comment]
Great! Let me know.

**Comment by danrr** [comment]
I tested uBlock, Adblock, Adblock Plus, and Ghostery on Chrome, Firefox, and Safari (all on Mac).
Only Safari triggered the `beforeload` event with `defaultPrevented` and it did it for all blockers. All other browsers just triggered an `error` event. Does that match what you were expecting?

**Comment by amorey** [comment]
Great, thanks for doing the research! I guess that answers the question about whether or not we can detect if a script was blocked by an ad blocker or if the download failed for another reason.

I'm still concerned about our ability to detect all blocked scripts as highlighted in this issue: https://github.com/muicss/loadjs/issues/47

**Comment by danrr** [comment]
I've not seen the issue described in #47. Loadjs always fired the correct success/error callback in my tests.
